### DISCUSSION TITLE
How to understand netNamespaceFilePath

### DISCUSSION THREAD

**Discussion by wangchao6856** [rook/rook#14540] [Q&A] [answered]
Regarding the netNamespaceFilePath parameter, based on the official documentation(`https://github.com/ceph/ceph-csi/blob/devel/examples/README.md`), this parameter has two main purposes:

Specifying the Network Namespace for Ceph-CSI: It allows you to specify the network namespace in which Ceph-CSI operates.

Ensuring Network Stability when Ceph-CSI Pods Encounter Issues: It helps maintain network stability in case of issues with Ceph-CSI Pods. To achieve this, you need a DaemonSet to persist the current network namespace and create a symbolic link from the `PID’s /proc/<PID>/ns/net` directory to `/var/lib/kubelet/plugins/<driver-name>/net`.

The question is, where should this DaemonSet be deployed? How do you create this directory link? Does it require manual handling?

**Comment by Madhu-1** [comment]
>The question is, where should this DaemonSet be deployed? How do you create this directory link? Does it require manual handling?

@wangchao6856 This is manual work if someone wants to run the cephcsi with pod networking (which is not encouraged due to the known problems)

if you are using please open the issue in cephcsi https://github.com/ceph/ceph-csi project.

**Reply by wangchao6856** [reply, marked as answer]
I removed the `netNamespaceFilePath` parameter, and after testing, the application's pod with the mounted Ceph PV was not affected when the Ceph-CSI pod was restarted
